lib/inferno/dsl/runnable.rb

```diff
@@ -464,6 +464,31 @@ def required_suite_options(suite_option_requirements)
           end
       end
 
+      # Replace a child test/group
+      #
+      # @param id_to_replace [Symbol, String] The ID of the child to be replaced.
+      # @param replacement_id [Symbol, String] The ID of the child that will take the
+      #   place of the child being replaced.
+      # @yield [Inferno::TestGroup, Inferno::Test] Optional block executed in the
+      #    context of the replacement child for additional configuration.
+      # @example
+      #   replace :test2, :test4 do
+      #     id :new_test_id
+      #     config(...)
+      #   end
+      def replace(id_to_replace, replacement_id, &)
+        index = children.find_index { |child| child.id.to_s.end_with? id_to_replace.to_s }
+        return unless index
```

Comment:
Let's make this a raise so that authors know immediately if they mess up the id.